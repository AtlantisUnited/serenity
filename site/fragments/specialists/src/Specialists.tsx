import React                        from 'react'

import { Avatar }                   from '@ui/avatar'
import { Button }                   from '@ui/button'
import { Carousel }                 from '@ui/carousel'
import { ChatIcon, StarIcon }       from '@ui/icons'
import { Image }                    from '@ui/image'
import { Box, Column, Layout, Row } from '@ui/layout'
import { NextLink as Link }         from '@ui/link'
import { Option, Search }           from '@ui/search'
import { Text }                     from '@ui/text'

import messages                     from './messages'

export const Specialists = ({
  intl,
  specialists = [],
  searchData = [],
  searchDefaultValue = '',
  onSearch,
  onContact,
}: any) => (
  <Column>
    <Layout flexBasis={48} alignItems='baseline'>
      <Text fontSize='massive' fontWeight='semiBold'>
        {intl.formatMessage(messages.specialists)}
      </Text>
      <Layout flexBasis={32} />
      <Text fontSize='huge' color='silver' fontWeight='medium'>
        {specialists.length}
      </Text>
    </Layout>
    <Layout flexBasis={20} />
    <Layout>
      <Search
        defaultValue={searchDefaultValue}
        placeholder={intl.formatMessage(messages.search)}
        showArrow={false}
        onSearch={onSearch}
      >
        {searchData.map((item) => (
          <Option key={item.id} value={item.name}>
            {item.name}
          </Option>
        ))}
      </Search>
    </Layout>
    <Layout flexBasis={7} />
    <Layout>
      <Column>
        {specialists.map((specialist) => (
          <Box
            key={specialist.id}
            my='7px'
            p='16px 25px 25px 16px'
            pr={['16px', '25px']}
            width='100%'
            border='black'
            borderRadius='extra'
            flexDirection='column'
            overflow='hidden'
          >
            <Layout>
              <Row>
                <Layout>
                  <Column>
                    <Layout>
                      <Avatar
                        width={56}
                        height={56}
                        color='alto'
                        src={(specialist.profile.photo && specialist.profile.photo.url) || null}
                      />
                    </Layout>
                    <Layout flexBasis={13} />
                    <Layout alignItems='center' justifyContent='center'>
                      <Layout>
                        <Text lineHeight='medium'>{specialist.rating}</Text>
                      </Layout>
                      <Layout pl='4px'>
                        <StarIcon width={16} height={16} />
                      </Layout>
                    </Layout>
                  </Column>
                </Layout>
                <Layout flexBasis={16} flexShrink={0} />
                <Layout flexGrow={1} overflow='hidden'>
                  <Column>
                    <Layout flexBasis='7px' />
                    <Layout>
                      <Row justifyContent='space-between'>
                        <Layout>
                          <Column>
                            <Layout>
                              <Link href={`/specialists/${specialist.id}`}>
                                <Text
                                  fontWeight='medium'
                                  fontSize='regular'
                                  lineHeight='medium'
                                  color='black'
                                >
                                  {specialist.interaction.formOfWork === 'company'
                                    ? specialist.interaction.name
                                    : `${specialist.profile.personalInformation.firstName} ${specialist.profile.personalInformation.lastName}`}
                                </Text>
                              </Link>
                            </Layout>
                            <Layout alignItems='center'>
                              <Layout>
                                <Text color='silver' lineHeight='medium' wordBreak='break-word'>
                                  {specialist.specialisation.main
                                    .map((item) => item.name)
                                    .join(', ')}
                                </Text>
                              </Layout>
                              <Layout>
                                {specialist.specialisation.additional &&
                                  specialist.specialisation.additional[0] && (
                                    <Box
                                      bg='alto'
                                      ml='12px'
                                      px='4px'
                                      height={20}
                                      alignItems='center'
                                      display='inline-block'
                                    >
                                      <Text
                                        color='silver'
                                        fontWeight='medium'
                                      >{`+${specialist.specialisation.additional.length}`}</Text>
                                    </Box>
                                  )}
                              </Layout>
                            </Layout>
                          </Column>
                        </Layout>
                        <Layout flexBasis={[0, 20]} flexShrink={0} />
                        <Layout alignItems='center' display={['none', 'none', 'flex']}>
                          <Button
                            borderRadius={20}
                            onClick={() =>
                              onContact({
                                id: specialist.id,
                                photo:
                                  (specialist.profile.photo && specialist.profile.photo.url) ||
                                  null,
                                name:
                                  specialist.interaction.formOfWork === 'company'
                                    ? specialist.interaction.name
                                    : `${specialist.profile.personalInformation.firstName} ${specialist.profile.personalInformation.lastName}`,
                              })
                            }
                          >
                            <ChatIcon width={20} height={20} color='rgb(153, 153, 153)' />
                            {intl.formatMessage(messages.contact)}
                          </Button>
                        </Layout>
                      </Row>
                    </Layout>
                    <Layout flexBasis={18} />
                    <Layout>
                      <Text lineHeight='medium'>
                        {(specialist.description || '').slice(0, 450)}
                      </Text>
                    </Layout>
                    <Layout flexBasis={specialist.portfolio[0] ? 20 : 0} />
                  </Column>
                </Layout>
              </Row>
            </Layout>
            <Layout flexBasis='5px' />
            <Layout ml={72} mr={-25} overflow='hidden'>
              <Layout ml={-16} width='100%'>
                <Carousel disableButton>
                  {specialist.portfolio.map(
                    (item) =>
                      item.images[0] && (
                        <Image
                          key={item.images[0].id}
                          width='auto'
                          height={120}
                          draggable='false'
                          mx={16}
                          src={item.images[0].url}
                        />
                      )
                  )}
                </Carousel>
              </Layout>
            </Layout>
          </Box>
        ))}
      </Column>
    </Layout>
    <Layout flexBasis={[40, 60, 80]} />
  </Column>
)
