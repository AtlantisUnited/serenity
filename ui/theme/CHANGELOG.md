# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 0.0.5 (2021-05-10)

**Note:** Version bump only for package @ui/theme

## 0.0.4 (2020-09-28)

### Bug Fixes

- **accounts:** add emotion-server, delete font ([08acf5a](https://github.com/Atlantis-Lab/serenity/commit/08acf5a7169352a606925fdfbbaa18f9917a8106))

## 0.0.3 (2020-06-30)

**Note:** Version bump only for package @ui/theme

## 0.0.2 (2020-06-10)

**Note:** Version bump only for package @ui/theme
