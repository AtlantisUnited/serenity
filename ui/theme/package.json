{
  "name": "@ui/theme",
  "version": "0.0.5",
  "license": "MIT",
  "main": "src/index.ts",
  "devDependencies": {
    "@types/react": "^17.0.6",
    "@types/styled-system": "5.1.4"
  },
  "dependencies": {
    "@emotion/css": "^11.1.3",
    "@emotion/react": "^11.4.0",
    "@emotion/styled": "^11.3.0",
    "react": "^17.0.2",
    "styled-system": "^5.1.0"
  }
}
