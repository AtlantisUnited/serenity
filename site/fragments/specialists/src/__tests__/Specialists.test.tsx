import React                     from 'react'
import { IntlProvider, useIntl } from 'react-intl'

import { MockedProvider }        from '@apollo/react-testing'
import { ThemeProvider }         from '@ui/theme'
import { render }                from '@testing-library/react'

import { Specialists }           from '../Specialists'

const TestComponent = () => {
  const intl = useIntl()

  return <Specialists intl={intl} />
}

describe('Specialists fragment', () => {
  describe('snapshots', () => {
    it('should match latest render snapshot', () => {
      const { asFragment } = render(
        <MockedProvider mocks={[]} addTypename={false}>
          <IntlProvider locale='ru' defaultLocale='ru' messages={{}}>
            <ThemeProvider>
              <TestComponent />
            </ThemeProvider>
          </IntlProvider>
        </MockedProvider>
      )

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
