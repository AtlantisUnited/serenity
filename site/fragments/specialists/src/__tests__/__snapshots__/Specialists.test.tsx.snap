// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Specialists fragment snapshots should match latest render snapshot 1`] = `
<DocumentFragment>
  .emotion-5 {
  font-family: 'Inter';
  font-weight: 600;
  font-size: 12px;
  width: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid rgba(0,0,0,0.08);
  background-color: #ffffff;
  outline: none;
}

.emotion-5 ul,
.emotion-5 li {
  margin: 0;
  padding: 0;
  list-style: none;
}

.emotion-5:hover,
.emotion-5.rc-select-focused {
  border: 1px solid rgba(0,0,0,0.2);
}

.emotion-5.rc-select-disabled {
  cursor: not-allowed;
}

.emotion-5.rc-select-disabled .rc-select-selection {
  opacity: 0.3;
  background-color: #dedede;
}

.emotion-5:hover.rc-select-disabled {
  border: 1px solid rgba(0,0,0,0.08);
}

.emotion-5 .rc-select-arrow {
  height: 100%;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  right: 16px;
}

@media (min-width:40em) {
  .emotion-5 .rc-select-arrow {
    right: 20px;
  }
}

.emotion-5 .rc-select-arrow svg {
  cursor: pointer;
}

.emotion-5 .rc-select-selection {
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.emotion-5 .rc-select-selection .rc-select-selection__rendered {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.emotion-5 .rc-select-selection .rc-select-selection__rendered .rc-select-selection__placeholder {
  position: absolute;
  top: 50%;
  font-weight: 500;
  color: #bebebe;
  cursor: text;
}

.emotion-5 .rc-select-selection .rc-select-selection__rendered ul {
  height: 100%;
  width: 100%;
}

.emotion-5 .rc-select-selection__choice {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #f3f3f3;
  border-radius: 4px;
  float: left;
  height: 30px;
  box-sizing: border-box;
  padding: 5px 10px 5px 15px;
  margin-right: 5px;
  margin-top: 5px;
  position: relative;
  overflow: hidden;
  cursor: auto;
}

.emotion-5 .rc-select-selection__choice .rc-select-selection__choice__remove {
  margin-left: 10px;
}

.emotion-5 .rc-select-selection__choice .rc-select-selection__choice__remove .rc-select-selection__choice__remove-icon {
  font-style: normal;
  color: #9e9e9e;
  cursor: pointer;
}

.emotion-5 .rc-select-selection__choice .rc-select-selection__choice__remove .rc-select-selection__choice__remove-icon:hover {
  color: #000000;
}

.emotion-5 .rc-select-search {
  width: 100%;
  height: 100%;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 .rc-select-search .rc-select-search__field {
  border: none;
  background: transparent;
  outline: 0px;
  width: 100%;
  height: 100%;
  padding: 0px;
  font-weight: 500;
  display: inline-block;
}

.emotion-5 .rc-select-search .rc-select-search__field__wrap {
  height: 100%;
  width: 100%;
  padding-top: 10px;
  padding-bottom: 10px;
  box-sizing: border-box;
}

.emotion-5 .rc-select-search .rc-select-search__field__mirror {
  position: absolute;
  top: -999px;
  left: 0;
  white-space: pre;
}

.emotion-5 .rc-select-selection-selected-value {
  opacity: 1 !important;
  position: absolute;
  top: 50%;
  margin-top: -7.5px;
}

.emotion-6 {
  min-height: 58px;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.03);
}

.emotion-6 .rc-select-selection {
  padding-left: 50px;
  padding-right: 24px;
}

@media (min-width:40em) {
  .emotion-6 .rc-select-selection {
    padding-left: 60px;
  }
}

.emotion-6 .rc-select-selection__placeholder {
  font-size: 13px;
  line-height: 1;
  margin-top: -6.5px;
}

.emotion-6 .rc-select-search__field {
  font-size: 13px;
}

.emotion-11 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 48px;
  -ms-flex-preferred-size: 48px;
  flex-basis: 48px;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.emotion-0 {
  font-size: 32px;
  font-weight: 600;
  font-family: Inter;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 32px;
  -ms-flex-preferred-size: 32px;
  flex-basis: 32px;
}

.emotion-2 {
  color: rgb(190,190,190);
  font-size: 24px;
  font-weight: 500;
  font-family: Inter;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 20px;
  -ms-flex-preferred-size: 20px;
  flex-basis: 20px;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  width: 100%;
}

.emotion-7 {
  height: 100%;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0;
  left: 16px;
}

@media (min-width:40em) {
  .emotion-7 {
    left: 20px;
  }
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 7px;
  -ms-flex-preferred-size: 7px;
  flex-basis: 7px;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 40px;
  -ms-flex-preferred-size: 40px;
  flex-basis: 40px;
}

@media screen and (min-width:40em) {
  .emotion-13 {
    -webkit-flex-basis: 60px;
    -ms-flex-preferred-size: 60px;
    flex-basis: 60px;
  }
}

@media screen and (min-width:52em) {
  .emotion-13 {
    -webkit-flex-basis: 80px;
    -ms-flex-preferred-size: 80px;
    flex-basis: 80px;
  }
}

<div
    class="emotion-11"
    display="flex"
    width="100%"
  >
    <div
      class="emotion-3"
      display="flex"
    >
      <span
        class="emotion-0"
        font-family="primary"
        font-size="massive"
        font-weight="semiBold"
      >
        Специалисты
      </span>
      <div
        class="emotion-1"
        display="flex"
      />
      <span
        class="emotion-2"
        color="silver"
        font-family="primary"
        font-size="huge"
        font-weight="medium"
      >
        0
      </span>
    </div>
    <div
      class="emotion-4"
      display="flex"
    />
    <div
      class="emotion-9"
      display="flex"
    >
      <div
        class="emotion-8"
      >
        <div
          class="emotion-5 emotion-6 rc-select rc-select-combobox rc-select-enabled rc-select-no-arrow"
        >
          <div
            aria-autocomplete="list"
            aria-controls="test-uuid"
            aria-expanded="false"
            aria-haspopup="true"
            class="rc-select-selection
            rc-select-selection--single"
            role="combobox"
          >
            <div
              class="rc-select-selection__rendered"
            >
              <div
                class="rc-select-selection__placeholder"
                style="display: block; user-select: none;"
                unselectable="on"
              >
                Поиск специалиста
              </div>
              <ul>
                <li
                  class="rc-select-search rc-select-search--inline"
                >
                  <div
                    class="rc-select-search__field__wrap"
                  >
                    <input
                      autocomplete="off"
                      class="rc-select-search__field"
                      value=""
                    />
                    <span
                      class="rc-select-search__field__mirror"
                    >
                       
                    </span>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="emotion-7"
        >
          <svg
            height="20"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
              stroke="none"
              stroke-width="1"
            >
              <g
                fill="#999999"
                fill-rule="nonzero"
                transform="translate(-4, -4)"
              >
                <path
                  d="M20.6666667,18.6666667 L19.6133333,18.6666667 L19.24,18.3066667 C20.5466667,16.7866667 21.3333333,14.8133333 21.3333333,12.6666667 C21.3333333,7.88 17.4533333,4 12.6666667,4 C7.88,4 4,7.88 4,12.6666667 C4,17.4533333 7.88,21.3333333 12.6666667,21.3333333 C14.8133333,21.3333333 16.7866667,20.5466667 18.3066667,19.24 L18.6666667,19.6133333 L18.6666667,20.6666667 L25.3333333,27.32 L27.32,25.3333333 L20.6666667,18.6666667 Z M12.6666667,18.6666667 C9.34666667,18.6666667 6.66666667,15.9866667 6.66666667,12.6666667 C6.66666667,9.34666667 9.34666667,6.66666667 12.6666667,6.66666667 C15.9866667,6.66666667 18.6666667,9.34666667 18.6666667,12.6666667 C18.6666667,15.9866667 15.9866667,18.6666667 12.6666667,18.6666667 Z"
                />
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
    <div
      class="emotion-10"
      display="flex"
    />
    <div
      class="emotion-9"
      display="flex"
    >
      <div
        class="emotion-11"
        display="flex"
        width="100%"
      />
    </div>
    <div
      class="emotion-13"
      display="flex"
    />
  </div>
</DocumentFragment>
`;
